components/Collections/examples/CollectionsSectionInsetsExample.m

```diff
@@ -0,0 +1,114 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+#import "CollectionsSectionInsetsExample.h"
+
+static const NSInteger kSectionCount = 5;
+static const NSInteger kSectionItemCount = 3;
+static NSString *const kReusableIdentifierItem = @"itemCellIdentifier";
+
+@implementation CollectionsSectionInsetsExample {
+  NSMutableArray *_content;
+  NSMutableDictionary *_sectionUsesCustomInsets;
+}
+
++ (NSArray *)catalogBreadcrumbs {
+  return @[ @"Collections", @"Custom Section Insets" ];
+}
+
++ (NSString *)catalogDescription {
+  return @"Demonstration of customizing section insets in Material Collections views.";
+}
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+  self.title = @"Custom Section Insets";
+  self.collectionView.accessibilityIdentifier = @"collectionsCustomSectionInsetsCollectionView";
+
+  // Register cell class.
+  [self.collectionView registerClass:[MDCCollectionViewTextCell class]
+          forCellWithReuseIdentifier:kReusableIdentifierItem];
+
+  // Populate content.
+  _content = [NSMutableArray array];
+  _sectionUsesCustomInsets = [NSMutableDictionary dictionaryWithCapacity:kSectionCount];
+  for (NSInteger i = 0; i < kSectionCount; i++) {
+    NSMutableArray *items = [NSMutableArray array];
+    for (NSInteger j = 0; j < kSectionItemCount; j++) {
+      NSString *itemString = [NSString stringWithFormat:@"Section-%zd Item-%zd", i, j];
+      [items addObject:itemString];
+    }
+    [_content addObject:items];
+  }
+
+  // Customize collection view settings.
+  self.styler.cellStyle = MDCCollectionViewCellStyleCard;
+}
+
+#pragma mark - <UICollectionViewDataSource>
```

Comment:
This entire section would be a good candidate to move to the -Supplemental file since the 'meat' of the instruction is the UICollectionViewDelegateFlowLayout section.